ui/src/components/AlertDialog/AlertDialog.tsx

```diff
@@ -16,30 +15,42 @@ import {
   InputAdornment,
 } from '@mui/material';
 import fetchAllContainers from '../../actions/fetchAllContainers';
-import { DDClient, DockerContainer, Alert } from '../../types';
-
-type AlertDialogProps = {
-  ddClient: DDClient;
-  dialogOpen: boolean;
-  setDialogOpen: React.Dispatch<React.SetStateAction<boolean>>;
-};
+import { DDClient, DockerContainer, UserAlert, DialogSettings } from '../../types';
 
 const metricHelperText = {
   'CPU %': '% of CPU used by the container based on the number of cores allocated',
   'MEM %': '% of memory used by the container',
 };
 
-export default function AlertDialog({ ddClient, dialogOpen, setDialogOpen }: AlertDialogProps) {
+const dialogTitle = {
+  NEW: 'Create Alert',
+  EDIT: 'Edit Alert',
```

Comment:
Is there a reason for the keys to be all uppercase here?

Reply:
The values on this object are accessed using `dialogSettings.mode` which is an alias for `<"NEW" | "EDIT">`